hstream/src/HStream/Server/Core/Cluster.hs

```diff
@@ -110,10 +110,10 @@ lookupSubscription sc req@LookupSubscriptionRequest{
     , lookupSubscriptionResponseServerNode     = Just theNode
     }
 
-{-# DEPRECATED lookupShardReader "Use lookupResource instead" #-}
 lookupShardReader :: ServerContext -> LookupShardReaderRequest -> IO LookupShardReaderResponse
```

Comment:
implement in `lookupResource`(ResShardReader) may be better?

Reply:
The problem is that the current code still contains the old shardReader logic and that part of the code needs to persist the results of the lookup. So to be compatible with both parts of the logic, either a new class of ResourceType should be added to the proto(https://github.com/hstreamdb/protocol/blob/2235b81637bce9f6bc46de9cf65e954977a326c0/hstream.proto#L568-L576), or a different lookup rpc should be used.

Reply:
if you want to keep the old shardReader, just don't change it, lookup shardReader will not affect the correction of readShardStream. `lookupResource and lookupShardReader` should always return same result and take same side effect in server until the old shardReader and `lookupShardReader` are removed.

Reply:
Ok, uniformly use lookupResource and persist the results of a lookup until the old reader is removed. Then we remove the persistent logic.